spec/tdd_spec.rb

```diff
@@ -0,0 +1,39 @@
+require_relative '../tdd'
+
+describe Solver do
+  solver = Solver.new
+  number = 5
+  three = 3
+  word = 'open'
+
+  context 'Test the solver class factoiral method.' do
+    it 'it should return the factorial of a number passed as an argument' do
+      factorial = solver.factorial(number)
+      expect(factorial).to eql 120
+    end

```

Comment:
- [ ] As required for the project, kindly ensure that The` #factorial` method has tests that handle the following cases:
  * Negative value which raises an exception as an expectation
  * Special case - when input is **0**, the result should equal **1**

Reply:
@tresorsawasawa thank you for letting me know about the short coming. I will make the required change.